vm/devices/storage/disk_backend/src/layered/resolve.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Resolver-related definitions for disk layer resources.
+
+use super::DiskLayer;
+use super::LayerIo;
+use vm_resource::kind::DiskLayerHandleKind;
+use vm_resource::CanResolveTo;
+
+impl CanResolveTo<ResolvedDiskLayer> for DiskLayerHandleKind {
+    type Input<'a> = ResolveDiskLayerParameters<'a>;
+}
+
+/// Parameters used when resolving a disk layer resource.
+#[derive(Copy, Clone)]
+pub struct ResolveDiskLayerParameters<'a> {
+    /// Whether the layer is being opened for read-only use.
+    pub read_only: bool,
+    #[doc(hidden)]
+    // Workaround for async_trait not working well with GAT input parameters
```

Comment:
meta: has this been fixed in newer async_trait versions, or is it still busted?

Reply:
Still busted

Reply:
womp womp

Reply:
We're most of all just stuck on rust-lang/rust#100013. Once that's fixed, we'll be able to move away from async_trait for this use case.